scripts/Muon/GUI/ElementalAnalysis2/grouping_widget/ea_grouping_tab_widget_presenter.py

```diff
@@ -0,0 +1,185 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantidqt.utils.observer_pattern import Observer, Observable, GenericObservable, GenericObserver
+from Muon.GUI.Common.utilities.run_string_utils import run_string_to_list
+
+
+class EAGroupingTabPresenter(object):
+    """
+    The grouping tab presenter is responsible for synchronizing the group table.
+    """
+
+    @staticmethod
+    def string_to_list(text):
+        return run_string_to_list(text)
+
+    def __init__(self, view, model,
+                 grouping_table_widget=None):
+        self._view = view
+        self._model = model
+
+        self.grouping_table_widget = grouping_table_widget
+
+        self._view.set_description_text(self.text_for_description())
+
+        # monitors for loaded data changing
+        self.loadObserver = EAGroupingTabPresenter.LoadObserver(self)
+        self.instrumentObserver = EAGroupingTabPresenter.InstrumentObserver(self)
+
+        # notifiers
+        self.groupingNotifier = EAGroupingTabPresenter.GroupingNotifier(self)
```

Comment:
I think we can replace these with the generic observables https://github.com/mantidproject/mantid/blob/b7e3f1ec4925addc6a09903b598612f6b10fdf76/qt/python/mantidqt/utils/observer_pattern.py#L85